package.json

```
@@ -5,9 +5,10 @@
 	"scripts": {
 		"build": "nx run-many --all --target=build",
 		"build:website": "nx build playground-website",
+		"build:remote": "nx build playground-remote",
 		"build:docs": "nx build docs-site",
 		"deploy:docs": "gh-pages -d dist/docs/build -t true",
-		"dev": "nx dev playground-website",
+		"dev": "nx dev playground-remote --host 0.0.0.0",
```

How would we develop the website version in this case?

---

That's a great question, and gets right to the heart of the matter. I needed remote.html to be served from my origin so that I could access the origin private file system. It wasn't clear to me how the build: website playground.wordpress.net worked at all. I could run npm run dev and it worked but I couldn't figure out how you got the index in to the production build. Adding my own index to the remote target let me deploy on cloudflare or even gh pages with a "normal" build. I would love to speak about this @bgrgicak . You are ahead of me in time and I have students now for the rest of the week. If we could speak next week, it would be amazing. Monday and Tuesday mornings EDT (Toronto time) are the best for me.

---

Aha, OPFS! I think you can only transfer OPFS handles between the same site, which typically also means the same origin. I remember failing to build an app that gets an OPFS handle on another domain and then gives playground.wordpress.net access to that handle.

I wonder what would happen if remote.html was responsible for getting the OPFS handle – in theory it would all stay within the same domain so perhaps that would be a way to enable custom Playground apps that are still able to use OPFS.

> Would it be acceptable to add a custom element to the client package? One would install the client package with npm and consume the custom element in a basic vite index.html.

Potentially, yes! I'm just not sure what do you mean by a custom element – would you elaborate?

---

> I would love to speak about this @bgrgicak

Please send me a message on the Making WordPress Slack (@bero) I'm happy to schedule some time to chat.

---

As a simple change, we could do this.

```suggestion
		"dev": "nx dev playground-website",
		"dev-remote": "nx dev playground-remote --host 0.0.0.0",
```